src/ucs/sys/topo/base/topo.c

```diff
@@ -809,10 +850,16 @@ double ucs_topo_get_pci_bw(const char *dev_name, const char *sysfs_path)
 const char *ucs_topo_resolve_sysfs_path(const char *dev_path, char *path_buffer)
 {
     const char *detected_type = NULL;
-    char device_file_path[PATH_MAX];
-    char *sysfs_realpath;
+    char *device_file_path    = NULL;
+    char *sysfs_realpath      = NULL;
+    char *sysfs_path          = NULL;
```

Comment:
pls try to avoid initialization and use goto/label to release what is needed

Reply:
done